etcdserver/server.go

```diff
@@ -2314,6 +2368,40 @@ func (s *EtcdServer) updateClusterVersion(ver string) {
 	}
 }
 
+func (s *EtcdServer) monitorDowngrade() {
+	t := s.Cfg.DowngradeCheckTime
+	if t == 0 {
+		return
+	}
+	lg := s.getLogger()
+	for {
+		select {
+		case <-time.After(t):
+		case <-s.stopping:
+			return
+		}
+
+		if !s.isLeader() {
+			continue
+		}
+
+		d := s.cluster.DowngradeInfo()
+		if !d.Enabled {
+			continue
+		}
+
+		targetVersion := d.TargetVersion
+		v := semver.Must(semver.NewVersion(targetVersion))
+		if isMatchedVersions(s.getLogger(), v, getVersions(s.getLogger(), s.cluster, s.id, s.peerRt)) {
+			lg.Info("the cluster has been downgraded", zap.String("cluster-version", targetVersion))
+			if _, err := s.downgradeCancel(context.Background()); err != nil {
```

Comment:
Is it safe to block forever with background context? If that's the case, can we comment?

Reply:
Changed to use context with timeout.